src/dmd/lambdacomp.d

```diff
@@ -0,0 +1,227 @@
+module dmd.lambdacomp;
+
+import core.stdc.stdio;
+import core.stdc.string;
+
+import dmd.declaration;
+import dmd.denum;
+import dmd.dsymbol;
+import dmd.expression;
+import dmd.func;
+import dmd.dmangle;
+import dmd.mtype;
+import dmd.root.outbuffer;
+import dmd.root.stringtable;
+import dmd.dscope;
+import dmd.statement;
+import dmd.tokens;
+import dmd.visitor;
+
+enum ExpType
+{
+    None,
+    EnumDecl,
+    Arg
+}
+
+extern (C++) class SerializeVisitor : SemanticTimeTransitiveVisitor
+{
+    alias visit = SemanticTimeTransitiveVisitor.visit;
+    OutBuffer buf;
+    StringTable arg_hash;
+    Scope* sc;
+    ExpType et;
+    Dsymbol d;
+
+    this(Scope* sc)
+    {
+        this.sc = sc;
+    }
+
+    override void visit(FuncLiteralDeclaration fld)
+    {
+        if (fld.type.ty == Terror)
+            return;
+
+        TypeFunction tf = cast(TypeFunction)fld.type;
+        uint dim = cast(uint)Parameter.dim(tf.parameters);
+        buf.printf("%d:", dim);
+
+        arg_hash._init(dim + 1);
+        foreach (i; 0 .. dim)
+        {
+            auto fparam = Parameter.getNth(tf.parameters, i);
+            if (fparam.ident !is null)
+            {
+                auto key = fparam.ident.toString().ptr;
```

Comment:
FWIW `.ptr` is deprecated in `@safe` code: https://dlang.org/changelog/pending.html#ptr-safe-end-of-deprecation